src/deploy/NetworkConfiguration.ts

```diff
@@ -100,110 +89,78 @@ interface TrackingInfo {
+    liquidationFactor: percentage(assetConfig.liquidationFactor),
+    supplyCap: number(assetConfig.supplyCap), // TODO: Decimals
+  }));
+}
+
+function getOverridesOrConfig(
+  overrides: ProtocolConfiguration,
+  config: NetworkConfiguration,
+  contracts: ContractMap,
+): ProtocolConfiguration {
+  const interestRateInfoMapping = (rates) => ({
+    supplyKink: _ => percentage(rates.supplyKink),
+    supplyPerYearInterestRateSlopeLow: _ => percentage(rates.supplySlopeLow),
+    supplyPerYearInterestRateSlopeHigh: _ => percentage(rates.supplySlopeHigh),
+    supplyPerYearInterestRateBase: _ => percentage(rates.supplyBase),
+    borrowKink: _ => percentage(rates.borrowKink),
+    borrowPerYearInterestRateSlopeLow: _ => percentage(rates.borrowSlopeLow),
+    borrowPerYearInterestRateSlopeHigh: _ => percentage(rates.borrowSlopeHigh),
+    borrowPerYearInterestRateBase: _ => percentage(rates.borrowBase),
   });
-}
-
-function getNetworkConfigurationFilePath(network: string): string {
-  return path.join(__dirname, '..', '..', 'deployments', network, 'configuration.json');
-}
-
-export async function hasNetworkConfiguration(network: string): Promise<boolean> {
-  let configurationFile = getNetworkConfigurationFilePath(network);
-  return await fileExists(configurationFile);
-}
-
-export async function loadNetworkConfiguration(network: string): Promise<NetworkConfiguration> {
-  let configurationFile = getNetworkConfigurationFilePath(network);
-  let configurationJson = await fs.readFile(configurationFile, 'utf8');
-  return JSON.parse(configurationJson) as NetworkConfiguration;
+  const trackingInfoMapping = (tracking) => ({
+    trackingIndexScale: _ => number(tracking.indexScale),
+    baseTrackingSupplySpeed: _ => number(tracking.baseSupplySpeed),
+    baseTrackingBorrowSpeed: _ => number(tracking.baseBorrowSpeed),
+    baseMinForRewards: _ => number(tracking.baseMinForRewards),
+  });
+  const mapping = () => ({
+    ...(config.governor && { governor: _ => address(config.governor) }),
+    ...(config.pauseGuardian && { pauseGuardian: _ => address(config.pauseGuardian) }),
+    symbol: _ => config.symbol,
+    baseToken: _ => getContractAddress(config.baseToken, contracts),
+    baseTokenPriceFeed: _ => address(config.baseTokenPriceFeed),
+    baseBorrowMin: _ => number(config.borrowMin), // TODO: in token units (?)
+    storeFrontPriceFactor: _ => percentage(config.storeFrontPriceFactor),
+    targetReserves: _ => number(config.targetReserves),
+    ...interestRateInfoMapping(config.rates),
+    ...trackingInfoMapping(config.tracking),
+    assetConfigs: _ => getAssetConfigs(config.assets, contracts)
+  });
+  return Object.entries(mapping()).reduce((acc, [k, f]) => {
+    return { [k]: overrides[k] ?? f(config), ...acc };
+  }, {});
 }
 
 export async function getConfiguration(
-  network: string,
-  hre: HardhatRuntimeEnvironment,
-  contractMapOverride?: ContractMap
+  deploymentManager: DeploymentManager,
+  contractOverrides?: ContractMap,
```

Comment:
Thoughts on setting the type of `contractOverrides` to `Map<string, string>` (map from alias to address) instead of `Map<string, Contract>`? We only ever use the address portion of the contracts and it's quite troublesome to set these overrides in a mainnet deploy script (we have to construct a contract instance for each token).

Another approach is to allow optionally setting the token addresses directly in `configuration.json`, which makes sense as well.

Reply:
Makes sense to me

Reply:
Should we do this in a later PR?

Reply:
Yeah, I can make the change in a separate PR. Ran into this issue while testing mainnet scenarios.